errors/src/main/java/com/palantir/conjure/java/api/errors/QosReason.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.api.errors;
+
+import com.google.errorprone.annotations.CompileTimeConstant;
+
+public final class QosReason {
+    public static final String DEFAULT_THROTTLE_REASON = "qos-throttle";
+    public static final String DEFAULT_RETRY_OTHER_REASON = "qos-retry-other";
+    public static final String DEFAULT_UNAVAILABLE_REASON = "qos-unavailable";
```

Comment:
Instead of making these public inside of QosReason, thought son making them private QosReason objects within each QosException subtype e.g. `private static final QosReason DEFAULT_REASON = QosReason.of("qos-throttle");

Reply:
Yea. I was deciding between the method you suggested, and having all of the reasons in QosReason. Ended up adding them to QosReason because it allowed me to view, and manage all the default values in one place. Given we likely won't be adding more exception types, I think your suggestion is reasonable as well.

Reply:
It's not so much about consolidating them for ourselves, but rather minimizing the API surface that we expose to library consumers. The less that they can interact with, the greater freedom we have to make changes in the future!